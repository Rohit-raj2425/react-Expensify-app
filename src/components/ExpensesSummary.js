import React from 'react';
import{ connect } from 'react-redux';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
 
export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenesWord =  expensesCount === 1 ? 'expense': 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {expensesCount} {expenesWord} totalling {formattedExpensesTotal}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);