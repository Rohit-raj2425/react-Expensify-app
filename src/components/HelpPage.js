import React from 'react';

const HelpPage = () => (
    <div>
        <p>This is from my help page!</p>
    </div>
);

export default HelpPage;