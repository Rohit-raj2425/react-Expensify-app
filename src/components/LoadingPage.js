import React from 'react';

export const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src="/images/loader.gif" />
    </div>
);

export default LoadingPage;