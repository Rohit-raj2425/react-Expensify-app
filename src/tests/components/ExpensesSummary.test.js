import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render expensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render expensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={235546677} />);
    expect(wrapper).toMatchSnapshot();
});