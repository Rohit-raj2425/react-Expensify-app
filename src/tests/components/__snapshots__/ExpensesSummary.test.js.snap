// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render expensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2,355,466.77
  </h1>
</div>
`;
